Source/Caches/HACache.swift

```diff
@@ -70,6 +70,32 @@ public class HACache<ValueType> {
         )
     }
 
+    /// Create a cache that relies on subscription updates without initial population.
+    ///
+    /// - Parameters:
+    ///   - connection: The connection to use and watch
+    ///   - subscribe: The info (one or more) for what subscriptions to start for updates or triggers for populating
+    public init(
+        connection: HAConnection,
+        subscribe: HACacheSubscribeInfo<ValueType>
```

Comment:
I think it should be possible to make only _this_ `ValueType` be Optional without affecting the rest of the subscription cases. Right now, we guarantee:

1. Populate produces a result T
2. Subscribe transforms a result T and returns a value type T, non-optional

The change to optionality happening elsewhere is now making it okay for a subscription to destroy the 'current' value. I think this is problematic because it gives an opportunity to accidentally set the wrong value; I can't really think of a reason we'd want to set 'current' to nil knowingly, outside of potentially this case.

This almost gets you there but requires figuring out how to store 'subscriptionInfo' (for tests to read) with alternate types, too. Maybe another variable?

```diff
diff --git a/Source/Caches/HACache.swift b/Source/Caches/HACache.swift
index d3adccc..3db665e 100644
--- a/Source/Caches/HACache.swift
+++ b/Source/Caches/HACache.swift
@@ -77,11 +77,11 @@ public class HACache<ValueType> {
     ///   - subscribe: The info (one or more) for what subscriptions to start for updates or triggers for populating
     public init(
         connection: HAConnection,
-        subscribe: HACacheSubscribeInfo<ValueType>
+        subscribe: HACacheSubscribeInfo<ValueType?>
     ) {
         self.connection = connection
         self.populateInfo = nil
-        self.subscribeInfo = [subscribe]
+        self.subscribeInfo = nil //[subscribe]
 
         self.start = { connection, cache, state in
             state.isWaitingForPopulate = false
@@ -368,7 +368,7 @@ public class HACache<ValueType> {
         subscription.start(connection, { [weak cache, weak connection] handler in
             guard let cache, let connection else { return }
             cache.state.mutate { state in
-                switch handler(state.current) {
+                switch handler(state.current!) {
                 case .ignore: break
                 case .reissuePopulate:
                     if let populate {
@@ -383,6 +383,33 @@ public class HACache<ValueType> {
         })
     }
 
+    private static func startSubscribe<ValueType>(
+        to subscription: HACacheSubscribeInfo<ValueType?>,
+        on connection: HAConnection,
+        populate: HACachePopulateInfo<ValueType>?,
+        cache: HACache<ValueType>
+    ) -> HACancellable {
+        subscription.start(connection, { [weak cache, weak connection] handler in
+            guard let cache, let connection else { return }
+            cache.state.mutate { state in
+                switch handler(state.current) {
+                case .ignore: break
+                case .reissuePopulate:
+                    if let populate {
+                        let populateToken = startPopulate(for: populate, on: connection, cache: cache)
+                        state.appendRequestToken(populateToken)
+                    }
+                case let .replace(value):
+                    state.current = value
+
+                    if let value {
+                        cache.notify(subscribers: state.subscribers, for: value)
+                    }
+                }
+            }
+        })
+    }
+
     /// Create a cancellable which removes this subscription
     /// - Parameter info: The subscription info that would be removed
     /// - Returns: A cancellable for the subscription info, which strongly retains this cache
diff --git a/Source/Caches/HACacheSubscribeInfo.swift b/Source/Caches/HACacheSubscribeInfo.swift
index 5202102..075b8e6 100644
--- a/Source/Caches/HACacheSubscribeInfo.swift
+++ b/Source/Caches/HACacheSubscribeInfo.swift
@@ -16,7 +16,7 @@ public struct HACacheSubscribeInfo<OutgoingType> {
     ///   - transform: The handler to convert the subscription's handler type into the cache's value
     public init<IncomingType: HADataDecodable>(
         subscription: HATypedSubscription<IncomingType>,
-        transform: @escaping (HACacheTransformInfo<IncomingType, OutgoingType?>) -> Response
+        transform: @escaping (HACacheTransformInfo<IncomingType, OutgoingType>) -> Response
     ) {
         let nonRetrySubscription: HATypedSubscription<IncomingType> = {
             var updated = subscription
@@ -27,7 +27,7 @@ public struct HACacheSubscribeInfo<OutgoingType> {
         self.init(
             request: subscription.request,
             anyTransform: { possibleValue in
-                guard let value = possibleValue as? HACacheTransformInfo<IncomingType, OutgoingType?> else {
+                guard let value = possibleValue as? HACacheTransformInfo<IncomingType, OutgoingType> else {
                     throw TransformError.incorrectType(
                         have: String(describing: possibleValue),
                         expected: String(describing: IncomingType.self)
@@ -65,12 +65,12 @@ public struct HACacheSubscribeInfo<OutgoingType> {
     ///   - current: The current value part of the transform info
     /// - Throws: If the type of incoming does not match the original IncomingType
     /// - Returns: The response from the transform block
-    public func transform<IncomingType>(incoming: IncomingType, current: OutgoingType?) throws -> Response {
-        try anyTransform(HACacheTransformInfo<IncomingType, OutgoingType?>(incoming: incoming, current: current))
+    public func transform<IncomingType>(incoming: IncomingType, current: OutgoingType) throws -> Response {
+        try anyTransform(HACacheTransformInfo<IncomingType, OutgoingType>(incoming: incoming, current: current))
     }
 
     /// The start handler
-    typealias StartHandler = (HAConnection, @escaping ((OutgoingType?) -> Response) -> Void) -> HACancellable
+    typealias StartHandler = (HAConnection, @escaping ((OutgoingType) -> Response) -> Void) -> HACancellable
 
     /// Type-erasing block to perform the subscription and its transform
     internal let start: StartHandler
diff --git a/Source/Caches/HACacheTransformInfo.swift b/Source/Caches/HACacheTransformInfo.swift
index e011ee7..f235d53 100644
--- a/Source/Caches/HACacheTransformInfo.swift
+++ b/Source/Caches/HACacheTransformInfo.swift
@@ -7,6 +7,6 @@ public struct HACacheTransformInfo<IncomingType, OutgoingType> {
 
     /// The current value of the cache
     /// For populate transforms, this is nil if an initial request hasn't been sent yet and the cache not reset.
-    /// For subscribe transforms, this is non-optional.
+    /// For subscribe transforms, this is nil if the populate did not produce results (or does not exist).
     public var current: OutgoingType
 }
diff --git a/Source/Caches/HACachedStates.swift b/Source/Caches/HACachedStates.swift
index 2759b0b..f16b3a0 100644
--- a/Source/Caches/HACachedStates.swift
+++ b/Source/Caches/HACachedStates.swift
@@ -9,14 +9,11 @@ private struct HACacheKeyStates: HACacheKey {
         .init(
             connection: connection,
             populate: .init(request: .getStates(), transform: { .init(entities: $0.incoming) }),
-            subscribe: .init(
-                subscription: .stateChanged(),
-                transform: { (info: HACacheTransformInfo<HAResponseEventStateChanged, HACachedStates?>) in
-                    guard var updated = info.current else { return .ignore }
-                    updated[info.incoming.entityId] = info.incoming.newState
-                    return .replace(updated)
-                }
-            )
+            subscribe: .init(subscription: .stateChanged(), transform: { info in
+                var updated = info.current
+                updated[info.incoming.entityId] = info.incoming.newState
+                return .replace(updated)
+            })
         )
     }
 }
diff --git a/Tests/HACache.test.swift b/Tests/HACache.test.swift
index d0aba82..207ae19 100644
--- a/Tests/HACache.test.swift
+++ b/Tests/HACache.test.swift
@@ -19,8 +19,10 @@ internal class HACacheTests: XCTestCase {
     private var populatePerform: (((CacheItem?) throws -> CacheItem) -> Void)?
 
     private var subscribeInfo: HACacheSubscribeInfo<CacheItem>!
+    private var subscribeInfoNoPopulate: HACacheSubscribeInfo<CacheItem?>!
     private var subscribeCancellableInvoked: Bool = false
     private var subscribePerform: (((CacheItem?) -> HACacheSubscribeInfo<CacheItem>.Response) -> Void)?
+    private var subscribePerformNoPopulate: (((CacheItem?) -> HACacheSubscribeInfo<CacheItem?>.Response) -> Void)?
     private var subscribeInfo2: HACacheSubscribeInfo<CacheItem>!
     private var subscribeCancellableInvoked2: Bool = false
     private var subscribePerform2: (((CacheItem?) -> HACacheSubscribeInfo<CacheItem>.Response) -> Void)?
@@ -91,6 +93,17 @@ internal class HACacheTests: XCTestCase {
                 }
             }
         )
+        subscribeInfoNoPopulate = HACacheSubscribeInfo<CacheItem?>(
+            request: .init(type: "none", data: [:]),
+            anyTransform: { _ in
+                fatalError()
+            }, start: { [weak self] _, subscribe in
+                self?.subscribePerformNoPopulate = subscribe
+                return HAMockCancellable { [weak self] in
+                    self?.subscribeCancellableInvoked = true
+                }
+            }
+        )
         subscribeInfo2 = .init(
             request: .init(type: "none", data: [:]),
             anyTransform: { _ in
@@ -675,7 +688,7 @@ internal class HACacheTests: XCTestCase {
     }
 
     func testInitSubscribingWithoutPopulate() {
-        let cache = HACache(connection: connection, subscribe: subscribeInfo)
+        let cache = HACache(connection: connection, subscribe: subscribeInfoNoPopulate)
         let result = cache.subscribe { _, _ in }
         XCTAssertEqual(cache.subscribeInfo?.count, 1)
         XCTAssertNil(cache.populateInfo)
diff --git a/Tests/HACacheSubscribeInfo.test.swift b/Tests/HACacheSubscribeInfo.test.swift
index 236d7dd..856d1f3 100644
--- a/Tests/HACacheSubscribeInfo.test.swift
+++ b/Tests/HACacheSubscribeInfo.test.swift
@@ -57,7 +57,7 @@ internal class HACacheSubscribeInfoTests: XCTestCase {
 
         let subscribeInfo = HACacheSubscribeInfo<SubscribeWrapper>(subscription: subscription, transform: { info in
             XCTAssertEqual(info.incoming, updated)
-            XCTAssertEqual(info.current?.subscribeItem, existing)
+            XCTAssertEqual(info.current.subscribeItem, existing)
             return result
         })
 
```